src/Workspaces/Core/Portable/Options/SerializableOptionSet.cs

```diff
@@ -133,8 +133,16 @@ private bool ShouldSerialize(OptionKey optionKey)
 
         public override OptionSet WithChangedOption(OptionKey optionKey, object? value)
         {
-            // make sure we first load this in current optionset
-            this.GetOption(optionKey);
+            // Make sure we first load this in current optionset
+            var currentValue = this.GetOption(optionKey);
+
+            // Check if the new value is the same as the current value.
+            if (value == currentValue)
+            {
+                // Return a cloned option set as the public API 'WithChangedOption' guarantees a new option set is returned.
```

Comment:
😕 I can see how the documentation reads this way, but does anyone actually depend on it? `SerializableOptionSet` is an immutable snapshot so I don't see how returning `this` would ever be a problem.

Reply:
I am not sure, I tried to keep the change as targeted as possible.